component/prometheus/scrape/scrape.go

```diff
@@ -174,11 +158,19 @@ func New(o component.Options, args Arguments) (*Component, error) {
 		opts:          o,
 		cluster:       clusterData,
 		reloadTargets: make(chan struct{}, 1),
-		scraper:       scraper,
-		appendable:    flowAppendable,
 		targetsGauge:  targetsGauge,
 	}
 
+	// Update created component with prometheus.Fanout and prometheus.ScrapeManager
+	data, err = o.GetServiceData(http.ServiceName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get information about HTTP server: %w", err)
+	}
+	httpData := data.(http.Data)
+	flowAppendable, scraper := c.createPromScrapeResources(httpData, args)
+	c.appendable = flowAppendable
+	c.scraper = scraper
```

Comment:
Is this necessary, if the same code will run in Update on line 175?

Reply:
I was trying to retain the previous setup. Now you're saying, it does look extra. Let see if the upcoming unit test can invalidate this step and I'll remove it